src/spec.c

```diff
@@ -1739,6 +1790,7 @@ FieldSpec *IndexSpec_CreateField(IndexSpec *sp, const char *name, const char *pa
   FieldSpec *fs = sp->fields + sp->numFields;
   memset(fs, 0, sizeof(*fs));
   fs->index = sp->numFields++;
+  RS_LOG_ASSERT(fs->index < RS_INVALID_FIELD_INDEX, "The maximum number of fields has been reached");
```

Comment:
is it possible here? aren't we failing to create the filed in this case? shouldn't we fail to create instead of crashing the server?